pkg/scalers/gitlab_runner_scaler.go

```diff
@@ -0,0 +1,206 @@
+package scalers
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/go-logr/logr"
+	"golang.org/x/sync/errgroup"
+	v2 "k8s.io/api/autoscaling/v2"
+	"k8s.io/metrics/pkg/apis/external_metrics"
+
+	"github.com/kedacore/keda/v2/pkg/scalers/scalersconfig"
+	kedautil "github.com/kedacore/keda/v2/pkg/util"
+)
+
+type pipelineStatus string
+
+const (
+	// pipelinePendingStatus is the status of the pending pipelines.
+	pipelinePendingStatus pipelineStatus = "pending"
+	// pipelineWaitingForResourceStatus is the status of the pipelines that are waiting for resources.
+	pipelineWaitingForResourceStatus pipelineStatus = "waiting_for_resource"
+	// pipelineRunningStatus is the status of the running pipelines.
+	pipelineRunningStatus pipelineStatus = "running"
+
+	// maxGitlabAPIPageCount is the maximum number of pages to query for pipelines.
+	maxGitlabAPIPageCount = 50
+	// gitlabAPIPerPage is the number of pipelines to query per page.
+	gitlabAPIPerPage = "200"
+)
+
+type gitlabRunnerScaler struct {
+	metricType v2.MetricTargetType
+	metadata   *gitlabRunnerMetadata
+	httpClient *http.Client
```

Comment:
Let's use https://github.com/kedacore/keda/blob/main/pkg/util/http.go

so we can controll http config globally

Reply:
Could you please elaborate? It seems like that's exactly the one being used (line 53)

https://github.com/kedacore/keda/pull/6499/files/b650bc3095169cba547a8a7bbafac4f511655724#diff-60834b5076af55bb90855035ef01a0e601f272fef17ae5c263e3a4abb348d27cR53

Reply:
cool, sorry overlooked this one :)